geps/gep-713/index.md

```diff
@@ -1,1157 +1,1082 @@
+    // TargetRefs specify the targets of the policy by name.
+    // The following kinds are supported: …
+    // +listType=map
+    // +listMapKey=group
+    // +listMapKey=kind
+    // +listMapKey=name
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []gatewayapiv1alpha2.LocalPolicyTargetReference `json:"targetRefs"`
+
+    // rest of the spec ("spec proper")…
+  }
+  ```
+</details>
+
+##### Label selectors
+
+The target reference includes the label selectors used to select a set of objects whose behavior to augment. E.g.:
 
 ```yaml
-apiVersion: networking.acme.io/v1alpha1
-kind: RetryPolicy
+apiVersion: policies.controller.io/v1
+kind: ColorPolicy
 metadata:
-  name: foo
+  name: my-color-policy
+spec:
+  targetRefs:
+  - group: gateway.networking.k8s.io/v1
+    kind: Gateway
+    selector: ## label selectors to a set of objects of the Gateway kind
+      matchLabels:
+        env: production
+  rules:
+    color: blue
+```
+
+<details>
+  <summary>Implementation tip</summary>
+
+  This targeting method can be implemented in Golang by using a type such as apimachinery's [`LabelSelector`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector) type. E.g.:
+
+  ```go
+  package color
+
+  import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+  )
+
+  type ColorPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the policy.
+    Spec ColorPolicySpec `json:"spec"`
+
+    // Status defines the current state of the policy.
+    Status ColorPolicyStatus `json:"status,omitempty"`
+  }
+
+  type ColorPolicySpec struct {
+    // TargetRefs specify the targets of the policy by label.
+    // The following kinds are supported: …
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []TargetRef `json:"targetRefs"`
+
+    // rest of the spec ("spec proper")…
+  }
+
+  type TargetRef struct {
+    // Group is the group of the target object.
+    Group Group `json:"group"`
+
+    // Kind is the kind of the target object.
+    Kind Kind `json:"kind"`
+
+    // Selector is the label selector of target objects of the specified kind.
+    Selector *metav1.LabelSelector `json:"selector"`
+  }
+  ```
+</details>
+
+
+##### Cross namespace references
+
+Metaresources can opt for allowing instances to target objects across Kubernetes namespaces, in which case an optional `namespace` field MUST be defined with the target reference.
+
+Although not strictly forbidden, this is in general discouraged due to [discoverability](#the-discoverability-problem) issues and security implications. Implementations that opt for designing metaresources that allow for cross namespace references should consider supporting [ReferenceGrants](https://gateway-api.sigs.k8s.io/api-types/referencegrant/?h=referencegrant) to address the security concern.
```

Comment:
I'd argue that if we want to allow cross-namespace references, we need at least one of the following to be true:

1. The policy is applied client-side and does not grant the client any additional access or permissions than it would otherwise have.
2. The policy is paired with ReferenceGrant or some other form of equivalent handshake that ensures that the target is accepting this policy. 

It's probably worth mentioning the confused deputy problem here for some context.